src/test/java/com/vinyl/vinylstore/LoginTest.java

```diff
@@ -57,19 +56,26 @@ public void tearDown() {
 	}
 
 	@Test
-	public void loginWithValidEmailAndPassword() {
+	public void loginWithValidEmailAndPasswordWhenTokenExists() {
 		Token token = createToken(user);
-
-		EmailPassDTO request = new EmailPassDTO();
-		request.setEmail(user.getEmail());
-		request.setPass(user.getPass());
+		setRequest(request);
 
 		ResponseEntity<TokenDTO> tdo = trt.postForEntity("/users/login", request, TokenDTO.class);
 
 		Assertions.assertThat(tdo.getStatusCode()).isEqualTo(HttpStatus.OK);
 		Assertions.assertThat(tdo.getBody().getHash()).isEqualTo(token.getHash());
 	}
 
+	@Test
+	public void loginWithValidEmailAndPasswordWhenTokenDoesNotExist() {
+		setRequest(request);
+
+		ResponseEntity<TokenDTO> tdo = trt.postForEntity("/users/login", request, TokenDTO.class);
+
+		Assertions.assertThat(tdo.getStatusCode()).isEqualTo(HttpStatus.OK);
+		Assertions.assertThat(tdo.getBody().getHash());
```

Comment:
you should actually do a find in the db and assert that the query is not null

Reply:
done